### PR TITLE
feat: price-feeder: telemetry additions

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Adds some measurements for Websocket reconnection attempts & candle / ticker messages being received

closes: #637

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/oracle/provider/binance.go [hunk 1]
```diff
@@ -237,6 +239,15 @@ func (p *BinanceProvider) messageReceived(messageType int, bz []byte) {
...
 	if len(tickerResp.LastPrice) != 0 {
 		p.setTickerPair(tickerResp)
+		telemetry.IncrCounter(
+			1,
+			"websocket",
+			"message",
+			"type",
+			"ticker",
+			"provider",
+			config.ProviderBinance,
```
Comment (adamewozniak) [inline price-feeder/oracle/provider/binance.go:249]: Just using the only constants that carry the names of the providers here - maybe we should have an additional set of constants in the `providers` package, but that seems a bit overkill for this

Comment (adamewozniak) [reply price-feeder/oracle/provider/binance.go:249]: Would also like @RafilxTenfen 's input on any other metrics that might be useful

**Comment by RafilxTenfen** [comment]
@adamewozniak 

The telemetry is going to increase the count when peggo uses the price-feeder? 
If so, do you think we should differentiate the usage with an key?

**Comment by RafilxTenfen** [comment]
What do you think about adding telemetry to `subscribedPairs` like:
```
       ....
       	p.setSubscribedPairs(cps...)
	telemetry.IncrCounter(
		float32(len(cps)),
		"websocket",
		"subscribe",
		"currency_pairs",
		"provider",
		config.ProviderBinance,
	)
```

**Comment by adamewozniak** [comment]
> @adamewozniak
> 
> The telemetry is going to increase the count when peggo uses the price-feeder? If so, do you think we should differentiate the usage with an key?

iirc it would only count if telemetry were enabled (e.g. if you called `telemetry.New`), so I don't really think we need a `price.feeder` key to differentiate, I don't think peggo has its own telemetry

**Review by RafilxTenfen — APPROVED** [review]

File: price-feeder/oracle/provider/gate.go [hunk 2]
```diff
@@ -197,6 +199,14 @@ func (p *GateProvider) SubscribeCurrencyPairs(cps ...types.CurrencyPair) error {
 		return err
 	}
 	p.setSubscribedPairs(cps...)
+	telemetry.IncrCounter(
+		float32(len(cps)),
+		"websocket",
+		"subscribe",
+		"currency_pairs",
+		"provider",
+		config.ProviderBinance,
```
Comment (toteki) [inline price-feeder/oracle/provider/gate.go:208]:
```suggestion
		config.ProviderGate,
```

File: price-feeder/oracle/provider/huobi.go [hunk 3]
```diff
@@ -157,6 +159,14 @@ func (p *HuobiProvider) SubscribeCurrencyPairs(cps ...types.CurrencyPair) error
 	}
 
 	p.setSubscribedPairs(cps...)
+	telemetry.IncrCounter(
+		float32(len(cps)),
+		"websocket",
+		"subscribe",
+		"currency_pairs",
+		"provider",
+		config.ProviderBinance,
```
Comment (toteki) [inline price-feeder/oracle/provider/huobi.go:168]:
```suggestion
		config.ProviderHuobi,
```

File: price-feeder/oracle/provider/gate.go (hunk 2 above)
Comment (adamewozniak) [reply price-feeder/oracle/provider/gate.go]: Knew I'd miss one

File: price-feeder/oracle/provider/kraken.go [hunk 4]
```diff
@@ -168,6 +170,14 @@ func (p *KrakenProvider) SubscribeCurrencyPairs(cps ...types.CurrencyPair) error
 	}
 
 	p.setSubscribedPairs(cps...)
+	telemetry.IncrCounter(
+		float32(len(cps)),
+		"websocket",
+		"subscribe",
+		"currency_pairs",
+		"provider",
+		config.ProviderBinance,
```
Comment (toteki) [inline price-feeder/oracle/provider/kraken.go:179]:
```suggestion
		config.ProviderKraken,
```

File: price-feeder/oracle/provider/okx.go [hunk 5]
```diff
@@ -170,6 +172,14 @@ func (p *OkxProvider) SubscribeCurrencyPairs(cps ...types.CurrencyPair) error {
 	}
 
 	p.setSubscribedPairs(cps...)
+	telemetry.IncrCounter(
+		float32(len(cps)),
+		"websocket",
+		"subscribe",
+		"currency_pairs",
+		"provider",
+		config.ProviderBinance,
```
Comment (toteki) [inline price-feeder/oracle/provider/okx.go:181]:
```suggestion
		config.ProviderOkx,
```

File: price-feeder/oracle/provider/coinbase.go [hunk 6]
```diff
@@ -211,6 +213,14 @@ func (p *CoinbaseProvider) SubscribeCurrencyPairs(cps ...types.CurrencyPair) err
 		return err
 	}
 	p.setSubscribedPairs(cps...)
+	telemetry.IncrCounter(
+		float32(len(cps)),
+		"websocket",
+		"subscribe",
+		"currency_pairs",
+		"provider",
+		config.ProviderBinance,
```
Comment (toteki) [inline price-feeder/oracle/provider/coinbase.go:222]:
```suggestion
		config.ProviderCoinbase,
```

**Review by toteki — APPROVED** [review]

**Pull request merged**